conan/internal/cache/db/table.py

```diff
@@ -10,20 +11,25 @@ class BaseDbTable:
     row_type: namedtuple = None
     columns: namedtuple = None
     unique_together: tuple = None
+    _lock: threading.Lock = None
+    _lock_storage = defaultdict(threading.Lock)
 
     def __init__(self, filename):
         self.filename = filename
         column_names: List[str] = [it[0] for it in self.columns_description]
         self.row_type = namedtuple('_', column_names)
         self.columns = self.row_type(*column_names)
+        self._lock = self._lock_storage[self.filename]
 
     @contextmanager
     def db_connection(self):
+        assert self._lock.acquire(timeout=10)
```

Comment:
Maybe it would be good to add some message in case it fails to acquire the lock?
```suggestion
        assert self._lock.acquire(timeout=10), "Conan failed to acquire database lock"
```